pkg/provision/provision.go

```diff
@@ -200,6 +208,67 @@ func Cluster(platform *platform.Platform) error {
 	return nil
 }
 
+// KindCluster provision or create a kubernetes cluster
+func KindCluster(platform *platform.Platform) error {
+	kubeadmPatches, err := createKubeAdmPatches(platform)
+	if err != nil {
+		return errors.Wrap(err, "failed to generate kubeadm patches")
+	}
+
+	kindConfig := ktypes.Cluster{
+		TypeMeta: ktypes.TypeMeta{
+			Kind:       "Cluster",
+			APIVersion: "kind.x-k8s.io/v1alpha4",
+		},
+		Nodes: []ktypes.Node{
+			{
+				Role: "control-plane",
+				ExtraPortMappings: []ktypes.PortMapping{
+					{
+						ContainerPort: 80,
+						HostPort:      80,
+						Protocol:      ktypes.PortMappingProtocolTCP,
+					},
+					{
+						ContainerPort: 443,
+						HostPort:      443,
+						Protocol:      ktypes.PortMappingProtocolTCP,
+					},
+				},
+				KubeadmConfigPatches: kubeadmPatches,
+			},
+		},
+	}
+
+	yml, err := yaml.Marshal(kindConfig)
+	if err != nil {
+		return errors.Wrap(err, "failed to marshal config")
+	}
+
+	tmpfile, err := ioutil.TempFile("", "kind.yml")
+	if err != nil {
+		return errors.Wrap(err, "failed to create tempfile")
+	}
+
+	if err := ioutil.WriteFile(tmpfile.Name(), yml, 0644); err != nil {
+		return errors.Wrap(err, "failed to write kind config file")
+	}
+
+	fmt.Printf("Config file: %s\n", tmpfile.Name())
+
+	if !platform.DryRun {
+		cmd := exec.Command("kind", "create", "cluster", "--config", tmpfile.Name())
```

Comment:
switch to `platform.GetBinary("kind")` which will handle downloading the binary if necessary - you can also execute the BinaryFunc in dry-run mode and it will just log what it would have executed.

in dry-run mode, we would want to print the contents of kind config file